### PR TITLE
PR dev branch Bubble Sort program

### PR DESCRIPTION
**Pull request by ad9311**
# Advanced Building Blocks - Bubble Sort
## Greetings! 👋🏽 

In this PR we present our Bubble Sort program that contains two methods as required. 
We have decided to use an integer array to showcase the results.  

This program was created by @fortuneonyeka and @ad9311

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,42 @@
...
+
+This project is part of the [Microverse](https://www.microverse.org/). This program compares one element to the one next to it and they are swapped if the one on the left is larger than the one on the right. This continues until the array is eventually sorted.
+
+### Built With
+
+- ![Ruby](https://img.shields.io/badge/-Ruby-red)
+
+### Getting Started
+
+If you want to check the source code out clone the repo. To know more about cloning and how git works check this page [Getting Started With Git](https://docs.github.com/en/github/getting-started-with-github/getting-started-with-git). But in short you can just clik the green button on top and download a .zip file, after decrompressing it, inside you will find a bubble_sort.rb.
```
Comment (Dipeshtwis) [inline README.md:13]: You can mention the command to clone and how to enter it into your folder before running your ruby file or rubocop.
```
$ git clone your-link.git
$ cd ruby_bubble_sort
```

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    array[i], array[i + 1] = array[i + 1], array[i] if yield(array[i], array[i + 1]).positive?
+  end
+  array
+end
+
+bubble_sort_by(numbers) do |left, right|
+  left.size - right.size
+end
+
+puts bubble_sort(numbers)
```
Comment (Dipeshtwis) [inline bubble_sort.rb:26]: Remove this line of code from your project.

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
# Status:  ♻️ Changes Required ♻️
 Hi @ad9311 and @fortuneonyeka

**You've really hit all the requirements in one shot. Well done.** Here're the major highlights of this PR:
- bubble_sort method implements the bubble sort algorithm.
- bubble_sort_by method is implemented and working

**However, in order to approve, we have to work a little bit more. Please check my comments below and fix them as appropriate.**

## For Approval

- [x] Do not commit debugging instructions in your code. Remove every other code that is not required in your project. Please see the in-line comments.
- [x] In your readme file, instead of using a statement to guide you can show with your code. Think about a person who is not from a technical background.


## Please make the required changes and submit another review request when you are done.

(Feel free to contact me on slack by the handle of  **deep** if you have any additional questions).
Happy Coding
[Dipesh Kumar](https://deeptwist.tk) | [Twitter](https://twitter.com/97deepeshkumar)

**Comment by ad9311** [comment]
#### UPDATE

The following changes were made as per requested:
- Remove from bubble_sort.rb unnecessary piece of code.
- Update README.md to have more helpful content.

**Review by mke2111 — APPROVED** [review]
# 😄 🥇  STATUS: APPROVED  🥇 👏🏽 

## Hey Happy Coders, 👋🏽 
Congratulations, 🥂 . You got minimum requirements for approval. 👍🏽 

### **_Well Done_** 💯 

    1. The readme file is clear and concise
    2. The linter checks run just fine
    3. The sorting methods do exactly just that.

### **_Room for Improvement_** 🤪 

    1. You may choose to remove the line `git checkout dev` from the getting started section of the readme file. Your app will mostly live in develop/main branch.

### You may now merge to the develop branch and remember to submit a project completion form.
### I wish you the very best in your upcoming projects.

**Cheers 🍹 and Happy coding 💻** 

<hr>

### Best regards
Slack 🆘  Roy Mukuye
Twitter 🐦  [Roy Mukuye](https://twitter.com/Roymkenya)
LinkedIn 🔗  [Roy Mukuye](https://www.linkedin.com/in/roy-mukuye/)

**Pull request merged**